noir-projects/aztec-nr/aztec/src/note/discovery/mod.nr

```diff
@@ -1,130 +1,77 @@
-use std::static_assert;
```

Comment:
Regarding filing structure, we've got discovery code living within `note/`, but discovery to me feels like a concept that shouldn't live within `note`, but should be "top level" under `src`.
It's similar to how log assembly isn't contained within `note/`, but lives under `src`.

As discussed earlier, partial note logs are actually better viewed as _event_ logs, so discovery of partial note (event) logs shouldn't live under `note/`.